### PR TITLE
implement genInsertable

### PR DESCRIPTION
**Pull request by danking**
Insertable types are those that can have .insert called.

In particular, we cannot insert into an optional struct
containing required fields because there is no sensible
defaults for those required fields.

I also did some clean up of the Type generators while
implementing this

### CODE REVIEW

**Comment by danking** [comment]
@jigold this will fix the bug that is surfacing in your PR

**Comment by danking** [comment]
cc: @cseed

File: src/main/scala/is/hail/expr/Type.scala [hunk 1]
```diff
@@ -22,39 +22,30 @@ import scala.reflect.classTag
...
-        genMaybeRequired.map { TSet(_) },
-        Gen.zip(genArb.resize(size - 1), genMaybeRequired).map { case (k, v) => TDict(k, v) },
-        genStruct.resize(size))
-    }
+  def genComplexType(required: Boolean) = {
+    val grDependents = GenomeReference.references.values.flatMap(gr =>
+      Array(TVariant(gr, required), TLocus(gr, required), TInterval(gr, required)))
+    val others = Array(
+      TAltAllele(required), TGenotype(required), TCall(required))
+    Gen.oneOfSeq(grDependents ++ others)
```
Comment (catoverdrive) [inline src/main/scala/is/hail/expr/Type.scala]: I'm not really sure what's happening here, but it's apparently returning an Iterable when the function expects a Seq.

Comment (danking) [reply src/main/scala/is/hail/expr/Type.scala]: blah

**Pull request merged**